enforcer/src/main/java/org/wso2/micro/gateway/enforcer/api/API.java

```diff
@@ -38,7 +39,14 @@
 
     default boolean executeFilterChain(RequestContext requestContext) {
         boolean proceed;
-        for (Filter filter : getFilters()) {
+        List<Filter> filterList = getFilters();
+        String lifeCycleState = requestContext.getMathedAPI().getAPIConfig().getApiLifeCycleState();
+        if (lifeCycleState.equals(APIConstants.PROTOTYPED_LIFE_CYCLE_STATUS)) {
```

Comment:
Move this initialization to the InitFilters method as it happens at the API's arrival point. Then we would not need to run this segment over each request.

Reply:
fixing with https://github.com/wso2/product-microgateway/pull/1682/commits/e8719a3b77a193a9c3053de0261b89eee8a8061c